File: webapp/schemas/connection.ts
import * as z from 'zod';

export const PROVIDER_SCHEMA = z.enum([
  'openai',
  'ollama',
  'xinference',
  'other',
]);

export const connectionSchema = z.object({
  id: z.number(),
  name: z.string().min(1, 'Connection name is required.'),
  provider: z.string(),
  api_base: z.string(),
  api_key: z.string().optional().nullable(),
  status: z.string().optional(),
  description: z.string().nullable().optional(),
  maintainer_id: z.number().optional(),
  created_at: z.string().optional(),
  updated_at: z.string().optional(),
  has_api_key: z.boolean().optional(),
  maintainer: z.object({ username: z.string() }).optional(),
}).passthrough()

export type Connection = z.infer<typeof connectionSchema>;

export const connectionCreateSchema = connectionSchema.omit({ 
  id: true,
  created_at: true,
  updated_at: true,
  maintainer_id: true,
});

export type ConnectionCreate = z.infer<typeof connectionCreateSchema>; 

File: backend/common/schemas/connection.py
from pydantic import BaseModel, Field, SecretStr
from typing import Optional, Literal
from datetime import datetime
from common.schemas.user import UserOut


class ConnectionBase(BaseModel):
    name: str = Field(..., description="连接名称")
    provider: Literal['openai', 'ollama', 'xinference', 'other'] = Field(..., description="模型提供商")
    api_base: str = Field(..., description="API Base URL")
    api_key: Optional[SecretStr] = Field(None, description="API Key")
    status: Optional[str] = Field("enabled", description="状态")
    description: Optional[str] = None
    maintainer_id: Optional[int] = None


class ConnectionCreate(ConnectionBase):
    api_key: Optional[str] = None


class ConnectionUpdate(ConnectionCreate):
    pass


class ConnectionConfig(BaseModel):
    base_url: str
    api_key: Optional[SecretStr] = None


class ConnectionTest(BaseModel):
    provider: str
    config: ConnectionConfig
    model_name: Optional[str] = None


class ConnectionOut(ConnectionBase):
    id: int
    has_api_key: bool = False
    maintainer: Optional[UserOut] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True 